src/commands/lambda/function.ts

```diff
@@ -96,38 +97,55 @@ export const getLambdaConfigs = async (
 
   const functionsToUpdate: FunctionConfiguration[] = []
 
-  for (const {config, functionARN} of results) {
-    const runtime = config.Runtime
-    if (!isSupportedRuntime(runtime)) {
-      throw Error(`Can't instrument ${functionARN}, runtime ${runtime} not supported`)
-    }
+  for (const config of results) {
+    const functionConfiguration = await getFunctionConfiguration(lambda, cloudWatch, config, region, settings)
+
+    functionsToUpdate.push(functionConfiguration)
+  }
 
-    await isLambdaActive(lambda, config, functionARN)
-    const lambdaLibraryLayerArn: string = getLayerArn(runtime, settings, region)
-    const lambdaExtensionLayerArn: string = getExtensionArn(settings, region)
-    const updateRequest = calculateUpdateRequest(
-      config,
-      settings,
-      lambdaLibraryLayerArn,
-      lambdaExtensionLayerArn,
-      runtime
-    )
-    let logGroupConfiguration: LogGroupConfiguration | undefined
-    if (settings.forwarderARN !== undefined) {
-      const arn = `/aws/lambda/${config.FunctionName}`
-      logGroupConfiguration = await calculateLogGroupUpdateRequest(cloudWatch, arn, settings.forwarderARN)
+  return functionsToUpdate
+}
+
+export const getLambdaConfigsFromRegEx = async (
+  lambda: Lambda,
+  cloudWatch: CloudWatchLogs,
+  region: string,
+  pattern: string,
+  settings: InstrumentationSettings
+): Promise<FunctionConfiguration[]> => {
+  const regEx = new RegExp(pattern)
+  const matchedFunctions: Lambda.FunctionConfiguration[] = []
+  let retryCount = 0
+  let listFunctionsResponse: Lambda.ListFunctionsResponse
+  let nextMarker: string | undefined
+  let firstCall = true
+
+  while (nextMarker || firstCall) {
```

Comment:
I think nextMarker + firstCall boolean are a bit redundant. I understand that you need to check whether or not this is the first call to handle the undefined marker possibility properly but it makes the code a bit harder to read.
Maybe you can wrap the listFunction to add a boolean which indicates whether or not an undefined value is OK (should be OK for the first call, but should stop the loop in other cases)

Reply:
Mmmm weird, I thought I had already solved this, maybe the merge stashed that code, I'll fix it again

Reply:
I do understand why it may seem redundant. The thing is that `nextMarker` can only be `undefined` at the beginning and in the end (when we've finished fetching all the functions). So if it's at the beginning, we do have to know if it's the first call to keep on retrying until it succeeds or it just reaches the retry limit.

Reply:
refactored using your recommendation from todays coding session